File: week1/src/basic_simple_talker.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <std_msgs/Int64.h>

int main(int argc, char** argv)
{
  ros::init(argc, argv, "basic_simple_talker");
  ros::NodeHandle nh;
  ros::Publisher chatter_pub = nh.advertise<std_msgs::Int64>("chatter", 10);
  ros::Rate loop_rate(10);

  while (ros::ok())
  {
    std_msgs::Int64 num;
    num.data = 12345;
    ROS_INFO("publish: %d", num.data);
    chatter_pub.publish(num);

    ros::spinOnce();
    loop_rate.sleep();
  }
  return 0;
}

File: src/week1/script/basic_simple_talker.py
#!/usr/bin/env python3
# license removed for brevity
import rospy
from std_msgs.msg import String
from std_msgs.msg import Int64

def talker():
     pub = rospy.Publisher('chatter', Int64, queue_size=10)
     rospy.init_node('talker', anonymous=True)
     r = rospy.Rate(10) # 10hz
     while not rospy.is_shutdown():
         num = 54321
         rospy.loginfo("publish:%d",num)
         pub.publish(num)
         r.sleep()
 
if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException: pass